benchmark/C/Savina/src/micro/Throughput.lf

```diff
@@ -0,0 +1,183 @@
+/**
+ * Micro-benchmark from the Savina benchmark suite.
+ * See https://shamsimam.github.io/papers/2014-agere-savina.pdf.
+ * 
+ * According to the Savina paper performance can be achieved by
+ * batch processing messages in the worker actors.
+ * This is not possible with reactors because scheduling and
+ * advancing logical time occurs after each round of
+ * message delivery.
+ * 
+ * @author Hannes Klein (c++ version)
+ * @author Matthew Chorlian (adapted C++ version to C)
+ */
+
+
+/* [[[cog
+# This file is a code generator using the python module cog:
+# See https://nedbatchelder.com/code/cog/
+#
+# All instructions for code  generation are in-lined in comments
+# like this one. With that you can use this file as a normal source file
+# but also to generate code.
+# 
+# To change the generated code in-line within this file run:
+# $ python -m cog -r this-file.lf
+# To generate a new file from this file stripping the generator code in the process run:
+# $ python -m cog -d -o output-file.lf this-file.lf
+#
+# Use the command line option -D to specify generator parameters, for example:
+# $ python -m cog -r -D parameter=100 this-file.lf
+#
+# Generator parameters used in this file:
+# -D numWorkers=60
```

Comment:
This benchmark should have two parameters: number of workers  and number of messages. Note that in the C++ version only the number of workers is a cog parameter, because the number of messages is implemented as a parameter to the main reactor. In C++, such parameters can be set when invoking the binary via command line arguments. This, however, is not possible in C and all parameters must be implemented as cog parameters.